services/venice-server/src/main/java/com/linkedin/venice/listener/ListenerService.java

```diff
@@ -150,6 +158,18 @@ public boolean startInner() throws Exception {
     serverFuture = bootstrap.bind(port).sync();
     LOGGER.info("Listener service started on port: {}", port);
 
+    grpcServer = new VeniceGrpcServer(
```

Comment:
IMO we should not create `VeniceGrpcServer ` object if `isGrpcEnabled` is set to false.
```java
if (isGrpcEnabled) {
    grpcServer = new VeniceGrpcServer( new VeniceGrpcServerConfig.Builder().setPort(grpcPort).setService(new VeniceReadServiceImpl()).build());)
}
```
Move `grpcServer` object creation on line number 154.

In `startInner` method you should only keep start logic and no instantiation 
```java
if (grpcServer != null) {
      grpcServer.start();
      LOGGER.info("gRPC service started on port: {}", grpcPort);
}
```